### ISSUE TITLE
Omni light  shadow softness exhibits  more aliasing than spot light soft shadows

### ISSUE THREAD

**Issue by Saul2022** [godotengine/godot#79347]
### Godot version

4.2 dev #61834  4.1 stable

### System information

Windows 11 amd radeon Tm graphics , 8gb ram, vulkan clustered

### Issue description

It seems like while  the spot light shadows  can benefit significantly from the softness filtering(pcf)  from this images, showing it correctly applied in the screenshots below.

Hard filter(no softness)

![editor_screenshot_2023-07-11T182327](https://github.com/godotengine/godot/assets/97898580/524eb2ed-f250-465f-b321-96f33efb41cd)

Soft low
![editor_screenshot_2023-07-11T183622](https://github.com/godotengine/godot/assets/97898580/1becc979-e656-40c8-8e69-3ee3a3e421c7)

Soft ultra 
![editor_screenshot_2023-07-11T183740](https://github.com/godotengine/godot/assets/97898580/62c2c634-12ab-4c3a-ba1f-e6c131ce16c7)

As seen the soft filter is well applied and it possitively affected on the spot light shadows, while in the case of the omni light shadows(cubemap), the pcs doesn´t make much effect on hiding the aliased edge, even though it 4048 resolution as seen.

Hard filter(no softness)

![editor_screenshot_2023-07-11T184230](https://github.com/godotengine/godot/assets/97898580/8fdc0143-cddc-46f7-9380-edeb8d2707ba)

Soft low

![editor_screenshot_2023-07-11T184319](https://github.com/godotengine/godot/assets/97898580/ba3bb396-a4aa-4e1a-bbd7-e7f8c8bc0af6)

Soft ultra

![editor_screenshot_2023-07-11T184402](https://github.com/godotengine/godot/assets/97898580/d49a80ab-a913-4a23-8d6e-2abbf2143da0)

Even though it applies here the smoothness it does not seem to hide the aliasing that are on the shadows  unlike the spotlight, it specially noticeable when you are close to  the shadows that it tries to hide it, but it doesn´t match at all .

### Steps to reproduce

1 Open the MRP

2 switch between enabling spot or omni lights  visibility to seeing their specific look, to not confuse between them.

3 You can compare the shadows look changing the filter sethings at the lights and shadows tab from the project  sethings.

### Minimal reproduction project

[shadows_test.zip](https://github.com/godotengine/godot/files/12017134/shadows_test.zip)

**Comment by Calinou** [comment]
Remember that you can rotate OmniLights to help hide artifacts due to shadows being rendered at oblique angles (relative to the shadow cubemap).

The same can be done with SpotLight's Z axis rotation.

**Comment by Saul2022** [comment]
> Remember that you can rotate OmniLights to help hide artifacts due to shadows being rendered at oblique angles 

I know, the problem that i see is that the  pcf on the omni shadow,  unlike the spot light  it does not seem  to hide the aliasing the shadows is emitting and yea while taa  shadow map jitter may help in future, it  can’t be the only solution if  it works on spot shadows.

**Comment by clayjohn** [comment]
Omnilights will always have more aliasing than SpotLights at the same resolution as they cover a much greater area. With spotlights, we can tightly fit the shadowmap around the bounds of the spotlight, while for Omnilights, the shadowmap has to cover the entire sphere around the light. 

That being said, Ultra quality should definitely remove aliasing artifacts and it appears from the screenshots that it doesn't. So it appears that something is going wrong in the Omnilight PCF calculations.

**Comment by Saul2022** [comment]
> That being said, Ultra quality should definitely remove aliasing artifacts and it appears from the screenshots that it doesn't. So it appears that something is going wrong in the Omnilight PCF calculations.

Ye, i notice the same on the low quality,  even if it  lower res than spot shadow, the smoothness can't seem to handle the small  aliasing( it inside the big aliasing  rectangle  while the spot light is almost able to handle it at the same softness quality

![IMG_20230712_100203](https://github.com/godotengine/godot/assets/97898580/8a6c1062-2c25-4d57-9572-872c26226adf)
![IMG_20230712_095944](https://github.com/godotengine/godot/assets/97898580/4fb55e53-cf44-4ac4-80b9-7ec648b37201)

**Comment by Calinou** [comment]
You can increase the **Shadow > Blur** property in the OmniLight to make the shadow softer, which should hide aliasing artifacts.

**Comment by Saul2022** [comment]
> You can increase the **Shadow > Blur** property in the OmniLight to make the shadow softer, which should hide aliasing artifacts

That true though sadly when blur is increasd performance goes worse aa seen in it explanation of the property, and can introduce sone patterm artifacts, depending on the shadow quality. 

Aside from that i think it could be good to check a bit on the wicked engine source code(as is MIT too, for the shadows, as it seems to handle them well with a 9 PCF sample. https://github.com/turanszkij/WickedEngine

**Comment by Calinou** [comment]
> That being said, Ultra quality should definitely remove aliasing artifacts and it appears from the screenshots that it doesn't. So it appears that something is going wrong in the Omnilight PCF calculations.

I think what's going on is just that Soft Ultra doesn't automatically increase the blur enough to hide shadow aliasing. If you increase blur, then the aliasing is pretty much gone.

**Comment by Saul2022** [comment]
> I think what's going on is just that Soft Ultra doesn't automatically increase the blur enough to hide shadow aliasing. If you increase blur, then the aliasing is pretty much gone.
> 
Not sure about that, does the spot light change their  blur value  depending on their softness on the code?  And it impact on performance?

**Comment by Calinou** [comment]
> Not sure about that, does the spot light change their blur value depending on their softness on the code? And it impact on performance?

All light types automatically adjust shadow blur behind the scene based on the shadow quality chosen. Higher qualities use a higher blur value to better make use of the greater sample count.

**Comment by Saul2022** [comment]
Further testing with calinou suggestions on increasing blur, most of them are on ultra soft quality

2.5 Blur


![editor_screenshot_2023-08-05T180122](https://github.com/godotengine/godot/assets/97898580/87518663-de28-4502-a3d2-4a5b70079a1d)


4 blur
![editor_screenshot_2023-08-05T180142](https://github.com/godotengine/godot/assets/97898580/57b1adcb-3bb7-466b-ad35-f5fe834eabf3)

And i used one with blur value 10 , but with medium sethings which made for me the most balanced results in stability and performance wise, though  it grainy, but calinou pr fixes that, so it ok.

![editor_screenshot_2023-08-05T180214](https://github.com/godotengine/godot/assets/97898580/df00b9fe-366b-4210-a23a-3143ae8ce6d1)

**Comment by Calinou** [comment]
Closing in favor of https://github.com/godotengine/godot-proposals/issues/840.

**Comment by Saul2022** [comment]
Revisiting this in 4.5 , i think the omni shadows have def improbed in quality ( not just the blur thanks to using taa but resolution itself. ,like this is 4k shadpwmap with soft low. Pretty well
Blur 1 
![Image](https://github.com/user-attachments/assets/2a209f1e-d319-4724-bf68-6bbb565cd3f5)

Blur 10 

![Image](https://github.com/user-attachments/assets/183d8188-469e-40a6-be99-bf0b04cc3ad1)

**Comment by Calinou** [comment]
I did a comparison on 4.6.dev2 (Forward+):

DirectionalLight3D | OmniLight3D | SpotLight3D
-|-|-
<img width="1152" height="648" alt="Image" src="https://github.com/user-attachments/assets/1c612db1-8bc5-4bd4-bdbb-081a41f02385" /> | <img width="1152" height="648" alt="Image" src="https://github.com/user-attachments/assets/a3fa20ea-d7f2-472b-a11e-a532dbd0eefa" /> | <img width="1152" height="648" alt="Image" src="https://github.com/user-attachments/assets/0bd7447b-8230-46d3-9faf-b1c21535f606" />

SpotLight3D | OmniLight3D with Blur = 2.0 and shadow atlas size = 8192
-|-
<img width="1152" height="648" alt="Image" src="https://github.com/user-attachments/assets/0bd7447b-8230-46d3-9faf-b1c21535f606" /> | <img width="1152" height="648" alt="Image" src="https://github.com/user-attachments/assets/f1cfcd10-c7b3-4047-91e2-f719fe7fa63c" />

We can't increase the default OmniLight3D shadow blur too much, as it would make the shadow look a lot blurrier and noisier than SpotLight3D out of the box. You can compensate for it with higher shadow atlas size, but this comes at a performance cost. However, something like https://github.com/godotengine/godot-proposals/issues/840 could allow you to have OmniLight3Ds that have shadows as sharp as SpotLight3Ds.

**Testing project:** [test_shadow_filter.zip](https://github.com/user-attachments/files/23059095/test_shadow_filter.zip)
*Press <kbd>Space</kbd> to switch cameras.*

**Comment by Saul2022** [comment]
> We can't increase the default OmniLight3D shadow blur too much,

That’s true, although using taa helps a lot in the case thanks to your original  pr that clay improved. I would say the blue doesn’t look as bad( atleast i think with the jitter  Patterm it far better to have a blur of 4 than seeing that stair  stepping.
